src/main/java/it/gov/pagopa/payhub/activities/service/exportflow/debtposition/BaseExportFlowFileService.java

```diff
@@ -0,0 +1,206 @@
+package it.gov.pagopa.payhub.activities.service.exportflow.debtposition;
+
+import it.gov.pagopa.payhub.activities.dto.export.ExportFileErrorDTO;
+import it.gov.pagopa.payhub.activities.dto.export.debtposition.ExportFlowFileResult;
+import it.gov.pagopa.payhub.activities.exception.exportFlow.ExportFlowFileNotFoundException;
+import it.gov.pagopa.payhub.activities.exception.exportFlow.InvalidExportFileException;
+import it.gov.pagopa.payhub.activities.service.CsvService;
+import it.gov.pagopa.payhub.activities.service.FileArchiverService;
+import it.gov.pagopa.payhub.activities.service.exportflow.ExportFileErrorArchiverService;
+import it.gov.pagopa.payhub.activities.util.Utilities;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+public abstract class BaseExportFlowFileService<E,F,D,C> {
+
+    private final CsvService csvService;
+    private final Class<C> csvRowDtoClass;
+    private final ExportFileErrorArchiverService exportFileErrorArchiverService;
+    private final FileArchiverService fileArchiverService;
+    private final Path workingDirectory;
+    private final String relativeFileFolder;
+    private final String fileNamePrefix;
+
+    protected BaseExportFlowFileService(CsvService csvService, Class<C> csvRowDtoClass, ExportFileErrorArchiverService exportFileErrorArchiverService, FileArchiverService fileArchiverService, Path workingDirectory, String relativeFileFolder, String fileNamePrefix) {
+        this.csvService = csvService;
+        this.csvRowDtoClass = csvRowDtoClass;
+        this.exportFileErrorArchiverService = exportFileErrorArchiverService;
+        this.fileArchiverService = fileArchiverService;
+        this.workingDirectory = workingDirectory;
+        this.relativeFileFolder = relativeFileFolder;
+        this.fileNamePrefix = fileNamePrefix;
+    }
+
+    public ExportFlowFileResult executeExport(Long exportFileId){
+        int[] pageNumber = {0};
+        long exportedRows = 0;
+        long errorCount = 0;
+        List<ExportFileErrorDTO> errorMessages;
+
+        E exportFileRecord = findExportFileRecord(exportFileId);
+        Long organizationId = getOrganizationId(exportFileRecord);
+        String profile = getFlowFileVersion(exportFileRecord);
+        Path csvFilePath = workingDirectory.resolve(String.valueOf(organizationId)).resolve(relativeFileFolder).resolve(getExportFileName(exportFileId));
+
+        if (checkStatusExportFileRecord(exportFileRecord)){
```

Comment:
don't ask to subclass to verify the status, the check is the same for all export flows... ask for the status and perform here (or in a private method) the check.
ps: if the export is not on the right status, let's throw an exception instead! Something like InvalidStatus..